front50-core/src/main/java/com/netflix/spinnaker/front50/model/plugins/remote/RemoteExtension.java

```diff
@@ -0,0 +1,70 @@
+package com.netflix.spinnaker.front50.model.plugins.remote;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.netflix.spinnaker.front50.model.plugins.PluginInfo;
+import com.netflix.spinnaker.front50.model.plugins.remote.stage.StageRemoteExtensionConfig;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/**
+ * A Spinnaker plugin's remote extension configuration.
+ *
+ * <p>This model is used by Spinnaker to determine which extension points and services require
+ * remote extension point configuration.
+ */
+@Data
+@NoArgsConstructor
+public class RemoteExtension {
+
+  /**
+   * The remote extension type. The remote extension is configured in the service that implements
+   * this extension type. The plugin release {@link PluginInfo.Release#requires} field is used to
+   * inform Spinnaker which service to use in configuring the remote extension type and if the
+   * remote extension is compatible with the running version of the Spinnaker service.
+   */
+  @Nonnull private String type;
+
+  /** Identifier of the remote extension. Used for observability. */
+  @Nonnull private String id;
+
+  /**
+   * Outbound transport configuration for the remote extension point; the protocol to address it
+   * with and the necessary configuration.
+   */
+  @Nonnull private RemoteExtensionTransport transport = new RemoteExtensionTransport();
+
+  /** Configures the remote extension point. */
+  @JsonTypeInfo(
+      use = JsonTypeInfo.Id.NAME,
+      include = JsonTypeInfo.As.EXTERNAL_PROPERTY,
+      property = "type")
+  @JsonSubTypes({@JsonSubTypes.Type(value = StageRemoteExtensionConfig.class, name = "stage")})
+  @Nullable
+  public RemoteExtensionConfig config;
+
+  /** Root remote extension configuration type. */
+  public interface RemoteExtensionConfig {}
```

Comment:
Rather than define the extension point config type in front50 (making front50 changes required for most remote extension), I think we should investigate a shared model or service-exposed schema for extension point configs - personally I favor the schema idea:

For any extension point that has a config, say `MyRemoteExtensionConfig`, we could convert to JSON Schema (or similar) and expose it from something like `/plugins/extension-point-configs/`. Each config would have a `type`, as you have it, but structured like `{service}/{configType}[@version?]`[1], allowing front50 to lookup the correct schema when a remote plugin is being published and we want to do some type / constraint validation. This would let us continue to define, in a single place, everything for an extension point.

I don't think we need to change the PR as it exists today, since this is talking about a lot of work - we can always retrofit what you have here after we've got something working end-to-end. 💯 

[1] Not sure if version is needed, but probably? 🤷 

One problem with this is that services tend to have dependencies _on_ front50, but not the other way around. I think I'd also be OK with services publishing their schemas into front50 on startup. 🤷

Reply:
Super interesting idea. 

I think a version would be needed.  Each change to the config schema would need to be versioned, and then that version would need to be the published version into the front50 plugin info release.  This is because we will be rolling out remote extensions by version, and we can't really swap out the underlying config schema without at least rolling out a new version.

Reply:
What would schema validation failures gate? Just publishes of new plugins?